src/commands/wallet/balance.rs

```diff
@@ -3,39 +3,75 @@ use clap::Args as ClapArgs;
 use eyre::ContextCompat;
 use std::sync::Arc;
 use zksync_ethers_rs::{
-    contracts::erc20::ERC20,
+    contracts::{erc20::ERC20, l2_shared_bridge::get_l2_token_from_l1_address},
+    core::utils::{format_ether, format_units},
     providers::{Middleware, Provider},
     types::Address,
+    utils::L2_ETH_TOKEN_ADDRESS,
+    ZKMiddleware,
 };
 
 #[derive(ClapArgs)]
 pub(crate) struct Args {
     #[clap(long = "token")]
     pub token_address: Option<Address>,
+    #[clap(long = "tol2", required = false)]
```

Comment:
Also, I'd suggest for this flag to require the `token_address` arg to make it impossible to run `zks wallet --l2` as this won't make sense since `zks wallet` itself returns the balance relative to the L2 by default (without the `--token` flag it'll return the L2 base token balance)

```suggestion
    #[clap(long = "tol2", required = false, requires = "token_address")]
```